tests/test_misc_common.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * -----------------------------------------------------------------------------
+ * test_misc_common.h --
+ *
+ * Header file with shared prototypes and definitions for miscellaneous
+ * functions that are shared between functional/ and unit/ test sources.
+ * This interface is provided so that stand-alone unit-tests that need these
+ * routines can be built w/o having to link-in all of SplinterDB.
+ *
+ * NOTE: Right now, we don't see a need for a .c file, which will require
+ *       listing that object in the Makefile for every unit-test that needs to
+ *       use these helper methods.
+ * -----------------------------------------------------------------------------
+ */
+#pragma once
+
+#include "util.h" // For STRING_EQUALS_LITERAL()
+
+/*
+ * Check if the user has supplied very 1st arg as "--use-shmem", which will
+ * run the test(s) using shared memory segment setup & allocation.
+ */
+static inline bool
+test_using_shmem(int argc, char *argv[])
```

Comment:
I'm confused.  Doesn't the generic config parsing already handle detecting that flag?

Reply:
Oh this is for the unit tests.

How about just making them all run all tests both w/ and w/o shmem?  No command-line parameter, which is the model we've been following so far for unit tests.

Reply:
@rtjohnso - This arg parsing and use of `config_parse()` v/s the newly added `test_using_shmem()` is a bit wonky. (I agree.)

config_parse does parse `--use-shmem`. But there is an ordering issue here in when that arg is parsed and when it has to be used.

A good example is the `tests/unit/btree_test.c` test. There we need to `platform_heap_create()` before the block of code that does other parsing and config-setup is done.

I _could_ rearrange this logic to have `config_parse()` do the required parsing. But that does not solve the problem entirely.

Take a look at `tests/unit/limitations_test.c`: There I just want to run this test with and w/o `--use-shmem` argument.

If I were to change that test to also now parse arguments using `config_parse()`, that will open up the gates to call this test with many more other parameters -- none of which are either meant for or tested-for in this test.

That's why I resorted to this `test_using_shmem()` approach, requiring that the `--use-shmem` arg be the very 1st arg to a unit-test, if you want to run it with shared memory enabled.

Reworking this entire argument parsing is quite a bit of work, and I really would not want to take it on now.

Reply:
And about: "How about just making them all run all tests both w/ and w/o shmem?"

You cannot just run a unit-test in both modes. To do that, we'd have to rework the unit-test execution machinery, something around the main() of ctests to run each test in both modes.

But then, not all tests are enabled to run w/ shared-memory. That's why you have to invoke them with the right argument.

The logic to execute is managed by commands in `test.sh`.

I think this test execution environment / usage is something we should take on far after all 3 parts of this feature dev is integrated to /main.

If I try to clean this up now, it will distract me further and will delay moving the remaining 2 parts forward.

Reply:
Discussed with @rtjohnso -- We agreed to leave this as is for now ... as other approaches will require quite a bit of updates to config_parse() and related machinery.

Reply:
After another round of review examining this code, we agreed to withdraw this newly added `test_using_shmem()` interface.

Deleted this special config parsing fn, and deleted this file.

Reworked all tests that were calling this function to now call `config_parse()` instead, to check if the test is being run with `--use-shmem` argument.